src/components/OneToDo/OneToDo.jsx

```diff
@@ -2,30 +2,77 @@ import { Checkbox, IconButton, TextField, Typography } from "@mui/material";
 import DeleteIcon from "@mui/icons-material/Delete";
 import EditIcon from "@mui/icons-material/Edit";
 import SaveIcon from "@mui/icons-material/Save";
-import React, { useRef } from "react";
+import React, { useEffect, useRef } from "react";
 import { useState } from "react";
 import "./OneToDo.css";
+import { useToDoList, editToDo, deleteToDo } from "../../store/store";
+import { useForm } from "react-hook-form";
+import { useMutation } from "@tanstack/react-query";
+import { deleteToDoFromJSON, putToDoInJSON } from "../../api/todoAPIs";
+
+function OneToDo({ index }) {
+  const editMutation = useMutation({
+    mutationFn: (editedToDo) => {
+      return putToDoInJSON({ id: editedToDo.id, newToDo: editedToDo });
+    },
+    mutationKey: ["editToDo"],
+  });
+
+  const deleteMutation = useMutation({
+    mutationKey: ["deleteToDo"],
+    mutationFn: (id) => deleteToDoFromJSON(id),
+  });
 
-function OneToDo({ index, toDoList, editToDo, deleteToDo }) {
   const [editingToDo, setEditingToDo] = useState(false);
   const [toDoStatus, setToDoStatus] = useState(false);
 
   const editField = useRef();
 
+  const toDoList = useToDoList().toDoList;
+
+  useEffect(() => {
```

Comment:
why are you using this state and the useEffect?
i mean we can directly use the todo.checked property